force-app/main/default/classes/SOQL.cls

```diff
@@ -31,7 +31,21 @@ public with sharing class SOQL {
     //[public | private ] [static] data_type method_name (input parameters) {
     //  The body of the method
     //}
+    
+    
+    public with sharing class LeadService {
+
+    public List<Lead> getAllLeads(){
+        // Check if the current user has read access to the Lead object
+        if (Schema.sObjectType.Lead.isAccessible()) {
```

Comment:
Unless we are doing specific error checking we normally don't worry about the accessible warnings that we get. It's cool that you found how to check for them though.